scripts/build/companion_libs/050-zlib.sh

```diff
@@ -84,6 +84,7 @@ do_zlib_backend() {
     CT_DoLog EXTRA "Configuring zlib"
 
     CT_DoExecLog CFG                                  \
+    CC="${host}-gcc"                                  \
```

Comment:
Can you check what was the contents of ` .build/${CT_TARGET}/buildtools/bin/x86_64-build_unknown-freebsd11.0-gcc` script when you had `gcc6` installed (and passed '6' as BUILD_SUFFIX)? It should be pointing to `/usr/local/bin/gcc6` in that case. If it does - it should be picked up by all the configure scripts by means of the filename `${host}-gcc`. If it does not - we have a bug somewhere in `scripts/crosstool-NG.sh.in` which creates these wrappers.